### ISSUE TITLE
every occurrence of run_once is marked as a failure against run_once[task]

### ISSUE THREAD

**Issue by LTek-online** [ansible/ansible-lint#3658]
##### Summary

Ansible lint marks every occurrence of the `run_once` keyword as a failure against the `run-once[task]` rule with the message that this task can cause issues if used with the `free` strategy **even if another strategy has been selected** or it is left to the default linear strategy.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
$ ansible-lint --version
ansible-lint 6.17.2 using ansible-core:2.15.2 ansible-compat:4.1.5 ruamel-yaml:0.17.32 ruamel-yaml-clib:0.2.7
```

OS: Debian 13
installed via pip3 in a virtualenv

##### STEPS TO REPRODUCE

Example: this simple playbook can trigger the failure:
```yaml
- 
- name: Test
  hosts: localhost
  connection: local
  gather_facts: false
  strategy: linear
  tasks:
    - name: Test
      ansible.builtin.debug:
      run_once: true

```

```console
$ ansible-lint 
WARNING  Listing 1 violation(s) that are fatal
run-once[task]: Using run_once may behave differently if strategy is set to free.
test.yaml:7 Task/Handler: Test

Read documentation for instructions on how to ignore specific rule violations.

              Rule Violation Summary               
 count tag            profile rule associated tags 
     1 run-once[task]         idiom                

Failed: 1 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'. Rating: 5/5 star

```

##### Desired Behavior

ansible-lint should not mark the `run_once` keyword as a failure when not using the free strategy. 

##### Actual Behavior

Regardless of which strategy is set in the playbook YAML or the ansible.cfg ansible-lint marks any occurrence of the `run_once`
keyword as a failure.

**Comment by ssbarnea** [comment]
Please read the documentation page https://ansible.readthedocs.io/projects/lint/rules/run-once/ - that behavior is clearly documented and reasoned there.
